### ISSUE TITLE
LibRdp 3.0.0     Android Client screen display problem (H264)

### ISSUE THREAD

**Issue by xiaoyvyv** [FreeRDP/FreeRDP#7669]
## LibRdp 3.0.0     Android Client screen display problem (H264)
> #### LibRDP 3.0.0-dev
> #### Source Code Release 2.6.0 

When H264 is on, the screen is black, and when H264 is off, the screen is displayed normally

### Conncet arguments （black screen）：
``` sh
[LibFreeRDP, /gdi:sw, /client-hostname:aFreeRDP-4f8aecdf-75db-4994-9cf, /v:xx.xx.108.0, /port:3389, /u:Administrator, /size:3110x1944, /bpp:32, /sec-rdp, /rfx, /gfx, /gfx:AVC444, -wallpaper, -window-drag, -menu-anims, +themes, -fonts, -aero, -glyph-cache, +async-channels, +async-input, +async-update, /clipboard, /audio-mode:0, /sound, /cert-ignore, /log-level:INFO]
```

**Comment by akallabeth** [comment]
How did you build?

**Comment by xiaoyvyv** [comment]
### VM ware，Ubuntu 18
- **Source Code Release 2.6.0**
Compile the FreeRDP binary library.

And import the project into Android Studio，build the APK。

 
![图片](https://user-images.githubusercontent.com/29088158/155483454-c12d1f68-0067-4d5e-8f01-c5760b66932c.png)

![图片](https://user-images.githubusercontent.com/29088158/155483680-ccac6c45-aee4-457d-a9c6-15fc71e47a09.png)

**Comment by akallabeth** [comment]
@xiaoyvyv ok, but the `how` refers to your build configuration. Which one did you use?
showing a screenshot that the build was successful is not useful information...

**Comment by xiaoyvyv** [comment]
```sh
./scripts/android-build-freerdp.sh --ndk $ANDROID_NDK --sdk $ANDROID_SDK
```

**Comment by xiaoyvyv** [comment]
#### The compile configuration file is like this ：
![图片](https://user-images.githubusercontent.com/29088158/155485562-031bffa9-1d6a-4c08-8c7f-2f92877d0f8d.png)

But the program supports H264 after running 
![图片](https://user-images.githubusercontent.com/29088158/155485950-dea38b4c-a03b-48a0-8908-4ee3cae4bce5.png)

**Comment by xiaoyvyv** [comment]
| | | |
|:-|:-|:-|
|![图片](https://user-images.githubusercontent.com/29088158/155487474-45a1f77e-59e8-4d7a-8aa3-7fce17b003fb.png) |![图片](https://user-images.githubusercontent.com/29088158/155487512-1d4dcfc2-9b6c-4113-81ff-8894aa431d39.png) |![图片](https://user-images.githubusercontent.com/29088158/155487542-4e7bdd79-126c-4da9-a839-d46a7bccf334.png) |

**Comment by akallabeth** [comment]
Well, if you don´t compile in any `H264` decoder, it will not decode `H264`...

**Comment by xiaoyvyv** [comment]
But `freerdp_has_h264 ` returns true, which should be a bug in the internal judgment of this method

**Comment by akallabeth** [comment]
@xiaoyvyv ah, right. there was recenty a pull request that added support to android mediacodec (`WITH_MEDIACODEC` is the option)
can´t help you there, this is the device hardware h264 decoder, if you don´t compile any other backend, then it can not decode if the hardware decoder does not support the settings requested

**Comment by hardening** [comment]
Yeah that's because it's a runtime detection, we try to load the mediacodec library, if it can setup everything then we use that decoder.

**Comment by akallabeth** [comment]
@hardening just checked it, this runtime detection is not a good idea, it hides that the backend is using features only available with API 28 and above without failing appropriately

**Comment by akallabeth** [comment]
Ok, fixed most of the stuff with #7671
